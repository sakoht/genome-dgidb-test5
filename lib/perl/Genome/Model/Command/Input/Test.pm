
###########################################################################

package Genome::Model::Command::Input::Test;

use strict;
use warnings;

use base 'Test::Class';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input';
}



###########################################################################

package Genome::Model::Command::Input::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;
require Genome::Model::Test;
require File::Temp;
require File::Path;

sub valid_param_sets {
    return grep { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub invalid_param_sets {
    return map { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_mock_model(
            type_name => 'tester',
            instrument_data_count => 0,
        )
            or confess "Can't create mock tester model.";
    }

    return $self->{_model};
}

sub _build {
    my $self = shift;

    return ($self->_model->builds)[0];
}



sub startup : Tests(startup => no_plan) {
    my $self = shift;
    
    no warnings;
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    use warnings;
    is_deeply(
        [ Genome::Model::Command->get_model_type_names ],
        [qw/ tester /],
        'Set tester type name.',
    );

    $self->_startup or confess "Failed startup.";
    
    return 1;
}

sub _startup {
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Add::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Add';
}

sub _valid_param_sets {
    return (
        { # add multiple 'value_ids' 
            name => 'friends',
            ids => 'Watson,Crick',
            before_execute => '_before_execute',
            after_execute => '_after_execute',
        },
        { # add single 'value'
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
            before_execute => '_before_execute',
            after_execute => '_after_execute',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to add a not 'is_many' input
            name => 'coolness', 
            ids => 'none',
        },
        { # try to add Crick again
            name => 'friends',
            ids => 'Crick',
        },
        { # try to add id that don't exist
            name => 'inst_data',
            ids => '2sep09.934noexist',
        },
    );
}

sub _before_execute {
    my ($self, $obj, $param_set) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    ok(!@ids, "No $name found for model.");
    
    return 1;
}

sub _after_execute {
    my ($self, $obj, $param_set) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    ok(@ids, "Added $name to model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Remove::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Genome::Utility::TestBase;
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Remove';
}

sub _valid_param_sets {
    return (
        { # remove multiple 'value_ids' 
            name => 'friends',
            ids => 'Watson,Crick',
        },
        { # remove single 'value'
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
            abandon_builds => 1,
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to remove a not 'is_many' input
            name => 'coolness', 
            ids => 'none',
        },
        { # try to remove input that is not linked to the model anymore
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
        },
    );
}

sub friends {
    return 'Crick';
}

sub inst_data {
    return Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
}

sub _startup {
    my $self = shift;

    my $model = $self->_model;

    $model->add_friend('Crick');
    $model->add_friend('Watson');
    my @friends = $model->friends();
    is_deeply(\@friends, [qw/ Crick Watson /], 'Added friends to remove.');
    my $id = Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
        or confess "Can't get sange id 2sep09.934pmaa1";
    $model->add_inst_data($id);
    my @inst_data = $model->inst_data;
    is_deeply(\@inst_data, [ $id ], 'Added instr_data to remove');
    
    my $build = $self->_build;
    my $build_input = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::Model::Build::Input',
        name => 'instrument_data',
        value_id => '2sep09.934pmaa1',
        build_id => $build->id,
    );
    $build->mock('abandon', sub{ return $build->build_event->event_status('Abandoned'); });

    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    #print Dumper(\@ids);
    ok(!@ids, "Removed $name from model.");
    if ( $obj->abandon_builds ) {
        is($self->_build->status, 'Abandoned', 'Successfully abandoned build');
    }

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::List::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::List';
}

sub valid_param_sets {
    return (
        {
            filter => 'model_id='.$_[0]->_model->id,
        },
    );
}

sub invalid_param_sets {
}

sub _startup {
    my $self = shift;

    $self->_model->coolness('low');
    is($self->_model->coolness, 'low', 'Set coolness (input)');

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Names::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Names';
}

sub valid_param_sets {
    return (
        {
            type_name => 'tester',
        },
    );
}

sub invalid_param_sets {
    return (
        {
            type_name => 'none',
        },
    );
}

sub startup : Tests(startup) {
    my $self = shift;
    
    no warnings 'once';
    no warnings 'redefine';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Update::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Update';
}

sub _valid_param_sets {
    return (
        { # remove single 'value'
            name => 'coolness',
            value => 'moderate',
        },
        { # remove single 'value'
            name => 'coolness',
            value => 'UNDEF',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to update an 'is_many' input
            name => 'friends',
            value => 'Watson',
        },
    );
}

###########################################################################

1;

=pod

=head1 Tests

=head1 Disclaimer

 Copyright (C) 2009 Washington University Genome Sequencing Center

 This script is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
 License for more details.

=head1 Author(s)

 Eddie Belter <ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

