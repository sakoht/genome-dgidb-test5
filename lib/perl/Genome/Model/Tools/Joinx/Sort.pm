package Genome::Model::Tools::Joinx::Sort;

use strict;
use warnings;

use Genome;
use Data::Dumper;

class Genome::Model::Tools::Joinx::Sort {
    is => 'Genome::Model::Tools::Joinx',
    has_input => [
        input_files => {
            is => 'Text',
            is_many => 1,
            doc => 'List of bed files to sort',
            shell_args_position => 1,
        },
    ],
    has_optional_input => [
        merge_only => {
            is => 'Boolean',
            value => '0',
            doc => 'If set, then the pre-sorted input files just merged',
        },
        output_file => {
            is => 'Text',
            doc => 'The output file (defaults to stdout)',
        },
    ],
};

sub help_brief {
    "Sorts one or more bed files."
}

sub help_synopsis {
    my $self = shift;
    return <<"EOS"
  gmt joinx sort a.bed [b.bed ...] --output-file sorted.bed
EOS
}

sub flags {
    my $self = shift;
    my @flags;
    push(@flags, "--merge-only") if $self->merge_only;
    return @flags;
}

sub execute {
    my $self = shift;
    my $output = "-";
    $output = $self->output_file if (defined $self->output_file);
    my @inputs = $self->input_files;
    my $flags = join(" ", $self->flags);
    my $cmd = $self->joinx_path . " sort $flags " .
        join(" ", @inputs) .
        " -o $output";


    my %params = (
        cmd => $cmd,
        # Sometimes these files come in empty in pipelines. We don't want this to choke when this happens, so don't check input file sizes.
        #input_files => \@inputs,
        allow_zero_size_output_files=>1,
    );
    $params{output_files} = [$output] if $output ne "-";
    Genome::Sys->shellcmd(%params);

    return 1;
}

1;
