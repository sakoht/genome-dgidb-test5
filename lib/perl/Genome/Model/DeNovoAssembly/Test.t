#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Data::Dumper 'Dumper';
use Test::More;

use_ok('Genome::Model::DeNovoAssembly::Test') or die;

my $rv;

#< PP >#
# fail - no seq platform
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model();
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - get w/o seq platform',
);
# fail - no assembler
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/o assembler',
);
# fail - invalid seq platform/assembler combo
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        assembler_name => 'consed',
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/ invalid seq paltform/assembler combo',
);

# valid velvet solexa
my $velvet_model = Genome::Model::DeNovoAssembly::Test->get_mock_model(
    assembler_name => 'velvet',
    sequencing_platform => 'solexa',
);
ok($velvet_model, 'Got mock DNA model for velvet solexa') or die;
my @inst_data = $velvet_model->instrument_data;
ok(@inst_data, 'Model instrument data');
ok(-s $inst_data[0]->archive_path, 'Solexa archive path exists');

# valid newbler 454
my $newbler_model = Genome::Model::DeNovoAssembly::Test->get_mock_model(
    assembler_name => 'newbler',
    sequencing_platform => 454,
);
ok($newbler_model, 'Got mock DNA model for newbler 454') or die;

done_testing();
exit;

#$HeadURL$
#$Id$
