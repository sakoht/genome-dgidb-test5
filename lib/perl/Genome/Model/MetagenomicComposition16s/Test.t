#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Data::Dumper 'Dumper';
use Test::More;

use_ok('Genome::Model::MetagenomicComposition16s::Test') or die;

my $rv;

#< PP >#
# fail - no seq platform
eval{
    $rv = Genome::Model::MetagenomicComposition16s::Test->get_mock_model();
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - get w/o seq platform',
);
# fail - invalid seq platform
eval{
    $rv = Genome::Model::MetagenomicComposition16s::Test->get_mock_model(
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/ invalid sequencing platform',
);

# valid 
my $model = Genome::Model::MetagenomicComposition16s::Test->get_mock_model(
    sequencing_platform => '454',
);
ok($model, 'Got mock DNA model') or die;
my @inst_data = $model->instrument_data;
ok(@inst_data, 'Model instrument data');

done_testing();
exit;

#$HeadURL$
#$Id$
