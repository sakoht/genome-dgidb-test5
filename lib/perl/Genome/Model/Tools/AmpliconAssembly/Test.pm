###########################################################################

package Genome::Model::Tools::AmpliconAssembly::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

use Data::Dumper 'Dumper';
require Genome::Model::Test;
use Test::More;

sub params_for_test_class {
    return (
        directory => $_[0]->amplicon_assembly->directory, # this creates the aa
        $_[0]->_params_for_test_class,
    );
}

sub amplicon_assembly {
    my $self = shift;

    unless ( $self->{_amplicon_assembly} ) {
        $self->{_amplicon_assembly} = Genome::Model::Tools::AmpliconAssembly::Set->create(
            directory => $self->tmp_dir,
        );
    }

    return $self->{_amplicon_assembly};
}

sub amplicons {
    return $_[0]->{_object}->get_amplicons;
}

sub _build_dir {
    return $_[0]->base_test_dir.'/Genome-Model-AmpliconAssembly/build-10000';
}

sub _params_for_test_class { return; }
sub should_copy_traces { 1 }
sub should_copy_edit_dir { 1 }
sub _pre_execute { 1 }

sub test01_copy_data : Tests {
    my $self = shift;

    if ( $self->should_copy_traces ) {
        ok( 
            Genome::Model::Test->copy_test_dir(
                $self->_build_dir.'/chromat_dir',
                $self->tmp_dir.'/chromat_dir',
            ),
            "Copy traces"
        ) or die;
    }

    if ( $self->should_copy_edit_dir ) {
        ok(
            Genome::Model::Test->copy_test_dir(
                $self->_build_dir.'/edit_dir',
                $self->tmp_dir.'/edit_dir',
            ),
            "Copy edit_dir"
        ) or die;
    }

    return 1;
}

sub test02_execute : Test(2) {
    my $self = shift;

    ok($self->_pre_execute, 'Pre Execute')
        or die "Failed method _pre_execute\n";

    ok($self->{_object}->execute, "Execute");
    #print $self->{_object}->directory,"\n"; <STDIN>;

    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::AssembleTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Assemble';
}

sub invalid_params_for_test_class {
    return (
        assembler_params => 'u-n-p-a-r-s able 00',
    );
}
sub _pre_execute {
    my $self = shift;

    my $amplicons = $self->amplicons;
    for my $amplicon ( @$amplicons ) {
        # remove ace files
        unlink $amplicon->ace_file;
    }

    my $cnt = grep { -s $_->ace_file } @$amplicons;
    die "Could not remove ace files\n" if $cnt;

    return 1;
}

sub test03_verify : Test(1) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $ace_cnt = grep { -s $_->ace_file } @$amplicons;
    is($ace_cnt, @$amplicons, 'Verified - Created an acefile for each amplicon');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::ClassifyTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Classify';
}

sub _pre_execute {
    my $self = shift;

    my $amplicons = $self->amplicons;
    for my $amplicon ( @$amplicons ) {
        my $class_file = $amplicon->classification_file;
        unlink $class_file if -e $class_file;
    }

    my $cnt = grep { -s $_->classification_file } @$amplicons;
    die "Could not remove classification files\n" if $cnt;

    return 1;
}

sub test03_verify : Test(1) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $cnt = grep { -s $_->classification_file } @$amplicons;
    is($cnt, @$amplicons, 'Verified - Created classification for each amplicon');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::ContaminationScreenTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::ContaminationScreen';
}

sub _params_for_test_class {
    return (
        database => '/gsc/var/lib/reference/set/2809160070/blastdb/blast',
        remove_contaminants => 1,
    );
}

sub _pre_execute { 
    # copy in the contaminated reads
    # contaminated read is vjn39g07.g1
    my $self = shift;

    for my $subdir ( (qw/ chromat_dir edit_dir /) ){
        my $source = $self->dir."/$subdir";
        my $dest = $self->tmp_dir."/$subdir";
        my $dh = Genome::Sys->open_directory($source)
            or die;
        while ( my $file = $dh->read ) {
            next if $file =~ m#^\.#;
            File::Copy::copy("$source/$file", $dest)
                or die "Can't copy ($source/$file) to ($dest): $!\n";
        }
    }

    $self->{_pre_execute_amplicons} = [ map { $_->get_name } @{$self->amplicons} ];

    return 1;
}

sub test03_verify : Test(2) {
    my $self = shift;

    # screen file
    my $screen_file = $self->{_object}->screen_file;
    ok(-s $screen_file, "Wrote screen file: $screen_file");

    # make sure we removed the conaminated amp
    my %pre_execute_amps = map { $_ => 1 } @{$self->{_pre_execute_amplicons}};
    for my $amplicon ( @{$self->amplicons} ) {
        delete $pre_execute_amps{$amplicon->get_name};
    }
    is_deeply(
        [ keys %pre_execute_amps ],
        [qw/ uyj40e03 /],
        'Removed contaminated amplicon: uyj40e03'
    );
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::CollateTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Collate';
}

sub test03_verify : Test(2) {
    my $self = shift;

    my $collate = $self->{_object};
    my @types = $collate->amplicon_assembly->amplicon_fasta_types;
    my $fasta_cnt = grep { -s $collate->amplicon_assembly->fasta_file_for_type($_) } @types;
    is($fasta_cnt, @types, 'Verified - Created a fasta for each type');
    my $qual_cnt = grep { -s $collate->amplicon_assembly->qual_file_for_type($_) } @types;
    is($qual_cnt, @types, 'Verified - Created a qual for each type');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::CopyFromBuild::Test;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Genome::Model::Test;
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::CopyFromBuild';
}

sub params_for_test_class {
    return (
        directory => $_[0]->tmp_dir,
        build_id => $_[0]->_model->last_complete_build_id,
        exclude_contaminated_amplicons => 1,
        copy_reads_only => 1,
    );
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_mock_model(
            type_name => 'amplicon assembly',
            use_mock_dir => 1,
        )
            or die "Can't create mock aa model";
    }

    return $self->{_model};
}

sub should_copy_traces { 0 }
sub should_copy_edit_dir { 0 }

sub _pre_execute {
    $File::Copy::Recursive::KeepMode = 0;
    return 1;
}

sub test03_verify : Test(4) {
    my $self = shift;

    my $amplicon_assembly = Genome::Model::Tools::AmpliconAssembly::Set->get(directory => $self->tmp_dir);
    ok($amplicon_assembly, 'Created amplicon assembly');
    ok($amplicon_assembly->exclude_contaminated_amplicons, 'Excluding contaminated amplicons');
    ok(scalar(glob($amplicon_assembly->chromat_dir.'/*')), 'Copied reads');
    ok(!scalar(glob($amplicon_assembly->edit_dir.'/*')), 'Did not copy edit dir');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::Create::Test;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Create';
}

sub params_for_test_class {
    return (
        directory => $_[0]->tmp_dir,
    );
}

sub should_copy_traces { 0 }
sub should_copy_edit_dir { 0 }

sub _pre_execute { 
    my $self = shift;

    die if -e $self->tmp_dir.'/properties.stor';
    
    return 1;
}

sub test03_verify : Test(1) {
    my $self = shift;

    ok(-e $self->tmp_dir.'/properties.stor', 'Created amplicon assembly');

    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::OrientTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Orient';
}

sub _pre_execute {
    my $self = shift;

    my $amplicons = $self->amplicons;
    for my $amplicon ( @$amplicons ) {
        my $ori_fasta = $amplicon->oriented_fasta_file;
        unlink $ori_fasta if -e $ori_fasta;
        my $ori_qual = $amplicon->oriented_qual_file;
        unlink $ori_qual if -e $ori_qual;
    }

    my $cnt = grep { -s $_->oriented_fasta_file } @$amplicons;
    die "Did not remove oriented fastas\n" if $cnt;

    return 1;
}

sub test03_verify : Test(2) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $fasta_cnt = grep { -s $_->oriented_fasta_file } @$amplicons;
    is($fasta_cnt, @$amplicons, 'Verified - Created oriented fasta for each amplicon');
    my $qual_cnt = grep { -s $_->oriented_qual_file } @$amplicons;
    is($qual_cnt, @$amplicons, 'Verified - Created oriented qual for each amplicon');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::PrepareData::Test;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::PrepareData';
}

sub should_copy_edit_dir { return; }

sub test03_verify : Test(1) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $fasta_cnt = grep { -s $_->fasta_file } @$amplicons;
    is($fasta_cnt, @$amplicons, 'Verified - Created fasta for each amplicon');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::TrimAndScreenTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

require File::Copy;
require File::Compare;
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::TrimAndScreen';
}

sub _params_for_test_class {
    return (
        trimmer_and_screener => 'trim3_and_crossmatch',
        #trimmer_and_screener_params => '-;',
    );
}

sub invalid_params_for_test_class {
    return (
        trimmer_and_screener => 'consed',
        trimmer_and_screener_params => '-;',
    );
}

sub _pre_execute {
    my $self = shift;

    my $amplicons = $self->amplicons;
    for my $amplicon ( @$amplicons ) {
        my $check_file = sprintf(
            '%s/%s.check.fasta',
            $amplicon->directory,
            $amplicon->name,
        );
        File::Copy::copy($amplicon->reads_fasta_file, $check_file)
            or die "Can't copy ".$amplicon->reads_fasta_file." to ".$check_file;
        unlink $amplicon->fasta_file;
        unlink $amplicon->qual_file;
        File::Copy::move($amplicon->reads_fasta_file, $amplicon->fasta_file)
            or die "Can't copy ".$amplicon->reads_fasta_file." to ".$amplicon->fasta_file;
        File::Copy::move($amplicon->reads_qual_file, $amplicon->qual_file)
            or die "Can't copy ".$amplicon->reads_fasta_file." to ".$amplicon->fasta_file;
    }

    return 1;
}

sub test03_verify : Test(1) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $compare_cnt = 0;
    for my $amplicon ( @$amplicons ) {
        my $pre_fasta = sprintf(
            '%s/%s.check.fasta',
            $amplicon->directory,
            $amplicon->name,
        );
        $compare_cnt++ if File::Compare::compare($amplicon->fasta_file, $pre_fasta);
    }
    is ($compare_cnt, scalar(@$amplicons), 'Trimmed and screened amplicons');

    return 1;
}

############################################################################

1;

