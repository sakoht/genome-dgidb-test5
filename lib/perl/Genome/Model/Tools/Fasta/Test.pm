package Genome::Model::Tools::Fasta::Trim::LucyTest;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

use File::Compare 'compare';
use Test::More;

sub lucy {
    return $_[0]->{_object};
}

sub test_class {
    return 'Genome::Model::Tools::Fasta::Trim::Lucy';
}

sub params_for_test_class {
    my $self = shift;
    return (
        vector_name => 'pCR4-TOPO',
        fasta_file => $self->fasta_file,
        output_fasta_file => $self->tmp_dir.'/lucy.fasta',
    );
}

sub required_attrs {
    return (qw/ vector_name fasta_file /);
}

sub fasta_file {
    return $_[0]->dir.'/H_KM-aab04h11.reads.fasta';
}

sub vector_fasta_file {
    return $_[0]->dir.'/pCR4-TOPO.fasta';
}

sub flanking_fasta_file {
    return $_[0]->dir.'/pCR4-TOPO.flanking.fasta';
}

sub test01_execute : Tests {
    my $self = shift;

    my $lucy = $self->lucy;

    # execute
    ok($lucy->execute, 'Executed lucy');

    # vector files
    diag( $self->vector_fasta_file.' => '.$lucy->vector_fasta_file );
    is( compare($self->vector_fasta_file, $lucy->vector_fasta_file), 0, 'Vector fasta file matches');
    diag( $self->flanking_fasta_file.' => '.$lucy->flanking_fasta_file );
    is( compare($self->flanking_fasta_file, $lucy->flanking_fasta_file), 0, 'Flanking fasta file matches');
    
    #print 'lucy => '.$lucy->_tmpdir,"\n".'fasta => '.$self->tmp_dir,"\n"; <STDIN>;

    return 1;
}

#####################################################################

package Genome::Model::Tools::Fasta::Trim::LucyReaderTest;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::Fasta::Trim::LucyReader';
}

sub test_class_sub_dir {
    return 'Genome-Model-Tools-Fasta-Trim-Lucy';
}
    
sub params_for_test_class {
    return (
        input => $_[0]->lucy_file,
    );
}

sub lucy_file {
        return $_[0]->dir.'/reader.debug.txt',
}

sub lucies_file {
        return $_[0]->dir.'/reader.lucies.stor',
}

sub test01_read_and_verify : Tests {
    my $self = shift;

    my @lucies = $self->{_object}->all;
    #print Dumper(\@lucies);
    #unlink $self->lucies_file; $self->store_file(\@lucies, $self->lucies_file);
    my $stored_lucies = $self->retrieve_file($self->lucies_file);
    is_deeply(\@lucies, $stored_lucies, 'Generated and stored lucies match');
    
    return 1;
}

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2008 Washington University Genome Sequencing Center

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

