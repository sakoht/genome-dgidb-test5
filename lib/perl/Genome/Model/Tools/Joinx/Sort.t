#!/usr/bin/env perl

use above 'Genome';
use Test::More tests => 4;
use File::Basename qw/dirname/;

my $pkg = 'Genome::Model::Tools::Joinx::Sort';
use_ok($pkg);

my $input = __FILE__ . ".input";
my $expected = __FILE__ . ".expected";
my $tmpdir = File::Temp::tempdir('joinx-sort-XXXXX', DIR => '/gsc/var/cache/testsuite/running_testsuites/', CLEANUP => 1);
my $output = join('/', $tmpdir, 'output');

my $cmd = $pkg->create(
    input_files => [$input],
    output_file => $output
    );

ok($cmd, 'Created command');
ok($cmd->execute, 'Executed command');

my $diff = Genome::Sys->diff_file_vs_file($output, $expected);
ok(!$diff, 'output matched expected result') or diag("diff results:\n" . $diff);

done_testing();
