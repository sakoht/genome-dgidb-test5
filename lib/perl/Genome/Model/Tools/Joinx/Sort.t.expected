1	112480082	112480083	C/C	0	0
1	243690673	243690674	C/C	0	0
2	38513160	38513161	T/T	0	0
3	112342877	112342878	T/T	0	0
3	147143428	147143429	G/G	0	0
4	55113050	55113051	C/C	0	0
4	100501787	100501788	C/C	0	0
5	38509905	38509906	C/C	0	0
6	117712683	117712684	T/T	0	0
7	51063507	51063508	C/C	0	0
7	156123074	156123075	C/C	0	0
8	128535995	128535996	C/C	0	0
10	12594196	12594197	C/Y	0	0
10	24742114	24742115	T/T	0	0
10	63503927	63503928	C/C	0	0
10	86803735	86803736	T/T	0	0
10	91727328	91727329	T/T	0	0
10	99206055	99206056	T/Y	0	0
10	105957613	105957614	C/C	0	0
11	2943275	2943276	C/C	0	0
11	21538240	21538241	C/T	0	0
12	67037613	67037614	C/C	0	0
13	60344809	60344810	G/G	0	0
14	64131440	64131441	C/C	0	0
16	456922	456923	T/T	0	0
16	79678853	79678854	C/Y	0	0
17	2844844	2844845	T/Y	0	0
19	36476434	36476435	T/Y	0	0
20	2025455	2025456	C/C	0	0
21	41965269	41965270	C/C	0	0
